bundles/org.openhab.binding.meteofrance/src/main/java/org/openhab/binding/meteofrance/internal/handler/RainForecastHandler.java

```diff
@@ -0,0 +1,166 @@
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.meteofrance.internal.config.ForecastConfiguration;
+import org.openhab.binding.meteofrance.internal.dto.RainForecast;
+import org.openhab.binding.meteofrance.internal.dto.RainForecast.Forecast;
+import org.openhab.binding.meteofrance.internal.dto.RainForecast.Properties;
+import org.openhab.binding.meteofrance.internal.dto.RainForecast.RainIntensity;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.TimeSeries;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RainForecastHandler} is responsible for updating channels
+ * and querying the API
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class RainForecastHandler extends BaseThingHandler implements MeteoFranceChildHandler {
+    private final Logger logger = LoggerFactory.getLogger(RainForecastHandler.class);
+    private final ChannelUID intensityChannelUID;
+    private final ZoneId systemZoneId;
+
+    private Optional<ScheduledFuture<?>> refreshJob = Optional.empty();
+    private Optional<PointType> location = Optional.empty();
+
+    public RainForecastHandler(Thing thing, ZoneId zoneId) {
+        super(thing);
+        this.intensityChannelUID = new ChannelUID(getThing().getUID(), INTENSITY);
+        this.systemZoneId = zoneId;
+    }
+
+    @Override
+    public @Nullable Bridge getBridge() {
+        return super.getBridge();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Rain Forecast handler.");
+        disposeJob();
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        ForecastConfiguration config = getConfigAs(ForecastConfiguration.class);
+        try {
+            PointType point = new PointType(config.location);
+
+            this.location = Optional.of(point);
+            this.refreshJob = Optional.of(scheduler.schedule(this::updateAndPublish, 2, TimeUnit.SECONDS));
+        } catch (IllegalArgumentException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Incorrect 'location' value.");
+        }
```

Comment:
Maybe minor, but in general it is best to keep the try/catch block as specific as possible.
```suggestion
        } catch (IllegalArgumentException e) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Incorrect 'location' value.");
            return;
        }
    this.location = Optional.of(point);
    this.refreshJob = Optional.of(scheduler.schedule(this::updateAndPublish, 2, TimeUnit.SECONDS));
```

Reply:
Ok, modified.